db/migrations/99_create_tables_for_batches_of_users.sql

```diff
@@ -0,0 +1,79 @@
+-- +migrate Up
+UPDATE `groups_login_prefixes`
+    SET `prefix` = CONCAT(`prefix`, '_')
+WHERE `prefix` NOT LIKE '%\_';
+
+CREATE TABLE `user_batch_prefixes` (
+    `group_prefix` VARCHAR(13) NOT NULL PRIMARY KEY COMMENT 'Prefix used in front of all batches',
+    `group_id` BIGINT(20) NOT NULL COMMENT 'Group and its subgroups in which managers can create users in batch',
```

Comment:
I've changed my mind, instead of blocking the group deletion (which may be complicated to understand from the UI) if it has batches (as `user_batches`  is restricted so that they do not disappear without having the possibility to remove the users), let's allow the `group_id` to be NULL and make the FK "ON DELETE SET NULL". 
You can also change the comment to 'Group (and its descendants) in which managers can create users in batch. NULL if the group was deleted, in which case batches should be cleaned manually."

Reply:
fixed